packages/router-plugin/src/serializer.ts

```diff
@@ -27,7 +27,7 @@ export class DefaultRouterStateSerializer
       fragment: route.fragment,
       data: route.data,
       outlet: route.outlet,
-      component: null,
+      component: route.component,
```

Comment:
This is a problem! We should never add a component to the state because this would cause the component to be made readonly by the dev mode's deepFreeze and result in runtime errors.
Maybe we could have a `componentType` property instead (set to the constructor ref) that is used for comparison.

Reply:
Hmmm... could the same type of component appear in multiple places in the RouterState?
If yes, then this is a much bigger problem to solve than just checking the type.

Reply:
Oh wait, I see that route.component has the type `component: Type<any> | string | null`.
This would not be the instance of the component but rather the Type.
Still would not want to freeze a type though. Maybe we could resolve this down to a string in the state if there is s type there?